surfsense_backend/app/services/document_processing/docling_service.py

```
@@ -0,0 +1,346 @@
+            self.converter = DocumentConverter(
+                format_options={
+                    InputFormat.PDF: pdf_format_option
+                }
+            )
+            
+            acceleration_type = "GPU (WSL2)" if self.use_gpu else "CPU"
+            logger.info(f"✅ Docling initialized successfully with {acceleration_type} acceleration")
+                
+        except ImportError as e:
+            logger.error(f"❌ Docling not installed: {e}")
+            raise RuntimeError(f"Docling not available: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)